### PR TITLE
Feature/#13/tow point cross over

### PR DESCRIPTION
**Pull request by tattoum**

### CODE REVIEW

File: srcs/Simulation.cpp [hunk 1]
```diff
@@ -145,7 +145,11 @@ Simulation::breedingSeason() {
   #pragma omp parallel for
   for (i = 0; i < _population.size(); ++i) {
     const couple_t couple = makeCouple(weightsSum);
-    const Individual child = crossOverSinglePoint(couple.first, couple.second).first;
+    std::cout << "ADN BEFOR CROSSOVER" << std::endl;
```
Comment (JeremyPouyet) [inline srcs/Simulation.cpp]: remove display!

File: srcs/Simulation.cpp [hunk 2]
```diff
@@ -181,10 +185,53 @@ couple_t Simulation::makeCouple(fitness_t weightsSum) {
...
+  {"TowPoint", &Simulation::crossOverTwoPoint},
+};
+
+couple_t Simulation::crossOverTwoPoint(const Individual &first, const Individual &second) {
+  int length = first.dnaSize() <= second.dnaSize() ? first.dnaSize() : second.dnaSize();
+  int firstPoint = RandomGenerator::getInstance().i_between(0, length);
+  int secondPoint = RandomGenerator::getInstance().i_between(0, length);
+  Individual newFirst, newSecond;
+
+
```
Comment (JeremyPouyet) [inline srcs/Simulation.cpp]: declare i at the begining of your function.

File: srcs/Simulation.cpp [hunk 3]
```diff
@@ -181,10 +185,53 @@ couple_t Simulation::makeCouple(fitness_t weightsSum) {
...
 
+const std::map<const std::string, const Simulation::func_ptr_t> Simulation::crossovers = {
+  {"SinglePoint", &Simulation::crossOverSinglePoint},
+  {"TowPoint", &Simulation::crossOverTwoPoint},
+};
+
+couple_t Simulation::crossOverTwoPoint(const Individual &first, const Individual &second) {
+  int length = first.dnaSize() <= second.dnaSize() ? first.dnaSize() : second.dnaSize();
+  int firstPoint = RandomGenerator::getInstance().i_between(0, length);
+  int secondPoint = RandomGenerator::getInstance().i_between(0, length);
```
Comment (JeremyPouyet) [inline srcs/Simulation.cpp]: replace 0 by firstpoint to avoid checking which point is the biggest.

**Review by JeremyPouyet — CHANGES_REQUESTED** [review]

**Review by tattoum — COMMENTED** [review]
change

**Pull request merged**